main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
+    with open("setup.json", encoding="utf-8") as setup_file:
+        settings = json.load(setup_file)
+
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
+    """Lifespan events.
+
+    This function is set as the lifespan of the application.
+
+    The code before 'yield' is executed once before the application starts.
+    The code after 'yield' is executed when the application is shutting down.
+    """
+    load_setup_file()
+    yield
+
+
+app = FastAPI(lifespan=lifespan)
+
+
+@app.get("/ls/")
```

Comment:
Which is the preferred style: `/ls` vs. `/ls/`? (just curious)

Reply:
Umm.. I'm also curious about it.

In my previous project, our team used the trailing slash so I just use it in this.
As I searched, there is no preferred style and it is up to the developer's preference.

I found two representative standard:
1. [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3) - "Uniform Resource Identifier (URI): Generic Syntax"
They doesn't use the trailing slash.
2. [W3C](https://www.w3.org/TR/url/)
They use the trailing slash.

So.. I think it is just our preference and I prefer it since the root url also has it. (http://127.0.0.1:8000/)

Reply:
Oh thank you for the research!